# IPL-workshop
Material for the Structural Bioinformatics workshop at Institut Pasteur Laos (Sept. 30 - Oct. 4 2024).

## Software required
Here is a list of software to be installed on your Linux/Mac laptop.
Some software require a (free) license:

* [Anaconda](https://www.anaconda.com/): management python modules
* [Python Notebook](https://jupyter.org): interactive environment to run python code
* [MDAnalysis](https://www.mdanalysis.org/): python library for structural analysis
* [ChimeraX](https://www.cgl.ucsf.edu/chimerax/): structure visualization
* [FoldX](https://foldxsuite.crg.eu/): scoring function for protein-protein interaction and more
* [DeepRank](https://pypi.org/project/deeprank/): scoring function for protein-protein interaction and more 
* [MODELLER](https://salilab.org/modeller/): homology modelling

## Online resources
These are the webservers and online resources we will use during the workshop:
* [BLAST](https://blast.ncbi.nlm.nih.gov/Blast.cgi): find similarities between sequences 
* [MAFFT](https://mafft.cbrc.jp/alignment/server/index.html): sequence alignment tool
* [AlphaFold2](https://colab.research.google.com/github/sokrypton/ColabFold/blob/main/AlphaFold2.ipynb): structure prediction from sequence
* [AlphaFold3](https://alphafoldserver.com/about): structure prediction from sequence
* [FoldSeek](https://search.foldseek.com/search): find similarities from structure
* [Pesto](https://pesto.epfl.ch/): parameter-free geometric deep learning method to predict protein interaction interfaces
* [ScanNet](http://bioinfo3d.cs.tau.ac.il/ScanNet/): geometric deep learning model for predicting protein binding sites from structure 
* [HADDOCK](https://www.bonvinlab.org/education/HADDOCK24/HADDOCK24-antibody-antigen-basic/): High Ambiguity Driven protein-protein DOCKing
* [SAbDab](https://opig.stats.ox.ac.uk/webapps/sabdab-sabpred/sabdab): database containing all the antibody structures available in the PDB

## Preliminary program
